File: Lesson2 Binary Search/BinarySearch.py
array = [1, 396, 99, 84, 33, 75, 9073]
array.sort()

key = 396

left = 0
right = len(array) - 1
found = False

while left <= right:
  mid = int((left + right) / 2)
  if array[mid] == key:
    print("Key found at index", mid)
    found = True;
    break;
  
  elif array[mid] > key:
    right = mid - 1
    
  elif array[mid] < key:
    left = mid + 1


if found == False:
  print("key not found in the array")

File: Lesson2 Binary Search/BinarySearch.cpp
#include<iostream>
#include<algorithm>

using namespace std;

int main()
{
    int array[] = {1, 396, 99, 84, 33, 75, 9073};
    sort(array, array  + 7);
    int left = 0, right = sizeof(array) / sizeof(array[0]) - 1;
    int key = 396;
    bool found = false;
    
    while(left <= right){
        int mid = (left + right) / 2;
        if(array[mid] == key){
            cout<<"key found at index "<<mid<<endl;
            found = true;
            break;
        }
        else if(array[mid] > key){
            right = mid - 1;
        }
        else if(array[mid] < key){
            left = mid + 1;
        }
    }
    
    if(found == false){
        cout<<"key not found in array"<<endl;
    }
   return 0;
}
